pycqed/analysis_v2/readout_analysis.py

```diff
@@ -1340,14 +1340,18 @@ class MultiQubit_SingleShot_Analysis(ba.BaseDataAnalysis):
             readouts for a single readout then n_readouts has to include them.
         channel_map: dictionary with qubit names as keys and channel channel
             names as values.
-        thresholds: dictionary with qubit names as keys and threshold values as
-            values.
+
     Optional options in the options_dict:
         observables: Dictionary with observable names as a key and observable
             as a value. Observable is a dictionary with name of the qubit as
-            key and boolean value indicating if it is selecting exited states.
+            key and boolean value indicating if it is selecting excited states.
             If the qubit is missing from the list of states it is averaged out.
         readout_names: used as y-axis labels for the default figure
+        thresholds: dictionary with qubit names as keys and threshold values as
+            values. If not specified, 'shots_thresholded' must be passed in.
+        shots_thresholded (dict): single shots thresholded keyed by qubit name.
```

Comment:
that is because the analysis only supports basic thresholding (one dimensional, on only one readout channel). When using 3 level readout or any other readout scheme with a more elaborate thresholding, the analysis would have to be changed to support all these different scenarios. If we end up finding that one of those is important enough to have it in the codebase, we can still add it later but we want to offer the flexibility to threshold the data outside (e.g. with gaussian mixutures for 3, 4, ... levels) and just pass in the thresholded shots.